// angular2-http/esm/src/angular2-http-redirect.service.js
import { Injectable } from '@angular/core';
import { Location } from '@angular/common';
import { Router } from '@angular/router';
var RedirectConfig = (function () {
    function RedirectConfig() {
    }
    return RedirectConfig;
}());
export { RedirectConfig };
export var RedirectConfigDefaults = {
    pathToLogin: '/login'
};
var RedirectService = (function () {
    function RedirectService(location, router, config) {
        this.location = location;
        this.router = router;
        this.config = config;
        this.redirectToUrlAfterLogin = {
            url: '/'
        };
        config = config || RedirectConfigDefaults;
    }
    RedirectService.prototype.restore = function () {
        this.router.navigate([this.redirectToUrlAfterLogin.url]);
    };
    RedirectService.prototype.save = function () {
        if (this.config.pathToLogin.toLocaleLowerCase() !== this.location.path().toLocaleLowerCase()) {
            this.redirectToUrlAfterLogin.url = this.location.path();
            this.router.navigate([this.config.pathToLogin]);
        }
    };
    return RedirectService;
}());
export { RedirectService };
RedirectService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
RedirectService.ctorParameters = function () { return [
    { type: Location, },
    { type: Router, },
    { type: RedirectConfig, },
]; };
